examples/src/main/resources/getGpuResources.sh

```diff
@@ -0,0 +1,33 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This is an example script that can be used to discover GPUs. It only works on NVIDIA GPUs since it
+# uses the nvidia-smi command. This script will find all visible GPUs, so if you aren't running
+# in an environment that can isolate GPUs to an executor and where multiple executors can run on a
+# single node you may not want to use this. See your cluster manager specific configs for other
+# options.
+#
+# It can be passed into Spark via the configs spark.executor.resource.gpu.discoveryScript and/or
+# spark.driver.resource.gpu.discoveryScript.
+# The script will return a JSON string in the format of the ResourceInformation class.
+#
+
+ADDRS=`nvidia-smi --query-gpu=index --format=csv,noheader | sed 'N;s/\n/\",\"/'`
```

Comment:
This doesn't work for me:

~~~
nvidia-smi --query-gpu=index --format=csv,noheader | sed 'N;s/\n/\",\"/'

0","1
2","3
4","5
6","7
~~~

It is fine to exclude it from this PR.

Reply:
let me make it more robust, I didn't test on system with 8 gpus, its simply an example script and not meant to cover all cases.

Reply:
How about leaving it to a follow-up PR?

Reply:
sure